src/azure/cli/tests/test_resource_id.py

```diff
@@ -35,15 +35,17 @@ def test_resource_id_complex(self):
         rtype = 'loadBalancers'
         bep = 'mybep'
         bep_type = 'backendAddressPools'
+        grandchild = 'grandchild'
+        gc_type = 'grandchildType'
```

Comment:
we should probably split this out into its own test since we lose coverage for child resources, which could be broken by a regex change